src/test/java/org/broadinstitute/hellbender/engine/dataflow/transforms/composite/AddContextDataToReadUnitTest.java

```diff
@@ -105,4 +107,63 @@ public void fullTest(List<GATKRead> reads, List<Variant> variantList,
         DataflowTestUtils.keyReadContextDataMatcher(result, pkvReadContextData);
         p.run();
     }
+
+    @DataProvider(name = "AddContextDataWithCustomReferenceWindowFunctionTestData")
+    public Object[][] addContextDataWithCustomReferenceWindowFunctionTestData() {
+        final String referenceBasesString = "AGCCTTTCGAACTGAGCCCGTTCCTGGGGTTATACCCGGCTTTTGGCGCT";
+        final RefAPISource mockReferenceSource = mock(RefAPISource.class, withSettings().serializable());
+
+        // Set up mock reference source to handle expected queries in our test cases below
+        // (will return null for unexpected queries)
+        for ( final SimpleInterval queryInterval : Arrays.asList(new SimpleInterval("1", 1, 10), new SimpleInterval("1", 1, 11), new SimpleInterval("1", 1, 15), new SimpleInterval("1", 20, 30), new SimpleInterval("1", 19, 31), new SimpleInterval("1", 17, 35), new SimpleInterval("1", 1, 40)) ) {
+            when(mockReferenceSource.getReferenceBases(any(PipelineOptions.class), any(RefAPIMetadata.class), eq(queryInterval))).thenReturn(new ReferenceBases(referenceBasesString.substring(queryInterval.getStart() - 1, queryInterval.getEnd()).getBytes(), queryInterval));
+        }
+
+        final List<Object[]> testCases = new ArrayList<>();
+        for ( final Class<?> readImplementation : Arrays.asList(SAMRecord.class, Read.class) ) {
+            // Test case layout: read, mock reference source, reference window function to apply, expected ReferenceBases for read
+            testCases.add(new Object[]{ makeRead("1", 1, 10, 0, readImplementation), mockReferenceSource, RefWindowFunctions.IDENTITY_FUNCTION, new ReferenceBases("AGCCTTTCGA".getBytes(), new SimpleInterval("1", 1, 10)) });
```

Comment:
Not sure what you mean here, but in general I think it's better to keep test data isolated rather than creating complex inter-dependencies across multiple test classes.